src/event-cache/EventCache.ts

```diff
@@ -220,12 +225,23 @@ export class EventCache {
             'aws:clientVersion': webClientVersion
         };
 
-        this.events.push({
-            details: JSON.stringify(eventData),
+        const partialEvent = {
             id: v4(),
-            metadata: JSON.stringify(metaData),
             timestamp: new Date(),
             type
+        };
+        this.eventBus.dispatch(Topic.EVENTS, {
+            ...(key && { key }),
```

Comment:
Note: `...(key && { key })` checks that key exists before adding property, avoiding dispatching the unnecessary `{key: undefined}`